extension/js/common/api/account-server.ts

```diff
@@ -60,44 +51,28 @@ export class AccountServer extends Api {
     recipients: ParsedRecipients,
     progressCb: ProgressCb
   ): Promise<UploadedMessageData> => {
-    if (await this.isFesUsed()) {
-      const fes = new EnterpriseServer(this.acctEmail);
-      // Recipients are used to later cross-check replies from the web
-      //   The message is not actually sent to them now.
-      //   Message is uploaded and a link is retrieved which is sent through Gmail.
-      return await fes.webPortalMessageUpload(encrypted, replyToken, from, recipients, progressCb);
-    } else {
-      return await FlowCryptComApi.messageUpload(await this.getIdToken(), encrypted, progressCb);
-    }
+    const service = await this.getExternalService();
+    return await service.webPortalMessageUpload(encrypted, replyToken, from, recipients, progressCb);
   };
 
   public messageGatewayUpdate = async (externalId: string, emailGatewayMessageId: string) => {
-    if (await this.isFesUsed()) {
-      const fes = new EnterpriseServer(this.acctEmail);
-      await fes.messageGatewayUpdate(externalId, emailGatewayMessageId);
-    }
+    const service = await this.getExternalService();
+    return await service.messageGatewayUpdate(externalId, emailGatewayMessageId);
   };
 
   public messageToken = async (): Promise<{ replyToken: string }> => {
-    if (await this.isFesUsed()) {
-      const fes = new EnterpriseServer(this.acctEmail);
-      return await fes.webPortalMessageNewReplyToken();
-    } else {
-      const res = await FlowCryptComApi.messageToken(await this.getIdToken());
-      return { replyToken: res.token };
-    }
+    const service = await this.getExternalService();
+    return await service.webPortalMessageNewReplyToken();
   };
 
   public isFesUsed = async (): Promise<boolean> => {
-    return await isFesUsed(this.acctEmail);
+    return await isCustomUrlFesUsed(this.acctEmail);
   };
```

Comment:
If you want to keep this method here (there's not much in it, could possibly be removed) then it should be also called `isCustomUrlFesUsed` instead of  `isFesUsed`

Reply:
ok